tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -253,6 +253,24 @@ def is_alive(self):
         return self.browser.is_alive()
 
 
+class TestSource:
+    def __init__(self, logger: structuredlog.StructuredLogger, test_queue: testloader.ReadQueue):
+        self.logger = logger
+        self.test_queue = test_queue
+        self.current_group = testloader.TestGroup(None, None, None, None)
+
+    def group(self) -> testloader.TestGroup:
+        if not self.current_group.group or len(self.current_group.group) == 0:
+            try:
+                self.current_group = self.test_queue.get()
+                self.logger.debug(f"Got new test group subsuite:{self.current_group[1]} "
+                                  f"test_type:{self.current_group[2]}")
+            except Empty:
+                self.logger.warning("Failed to get group from queue")
```

Comment:
Should we still have this warning if an empty queue no longer represents an anomaly? This was removed in https://github.com/web-platform-tests/wpt/pull/42350#discussion_r1350547682 for being too noisy.

Reply:
I deleted that and didn't save the file. Sigh :)